runtime/src/iree/tooling/trace_replay.c

```diff
@@ -560,15 +581,12 @@ static uint8_t iree_trace_replay_pseudorandom_uint8(uint32_t* state) {
 static void iree_trace_replay_generate_fully_specified_pseudorandom_buffer(
     iree_hal_element_type_t element_type, iree_byte_span_t span,
     uint32_t seed) {
-  const bool is_unsigned = iree_hal_element_numerical_type(element_type) ==
-                           IREE_HAL_NUMERICAL_TYPE_INTEGER_UNSIGNED;
   iree_host_size_t element_byte_count =
       iree_hal_element_dense_byte_count(element_type);
   uint8_t* data_end = span.data + span.data_length;
   uint32_t state = seed;
   for (uint8_t* data = span.data; data < data_end; data += element_byte_count) {
-    int value_in_uint8_range = iree_trace_replay_pseudorandom_uint8(&state);
-    int value = value_in_uint8_range + (is_unsigned ? 0 : -128);
+    double value = iree_trace_replay_pseudorandom_double(&state);
```

Comment:
I don't feel entirely good about routing all element types through `double`. I feel that it would be nice to only use floating point at all (resp. `double`) when `element_type` is a floating-point type (resp. is `double`). Reasons:
* Floating point will always be potentially weird, in terms of potentially having unexpected hardware floating-point environment/state interfere with results. When `element_type` is integral, such weirdness would be particularly unexpected.
* On some targets, floating-point is very slow (e.g. when it's emulated in software). We care about e2e tests running not much slower than necessary there.
* Even on targets that have fast `float`, `double` may still be slow.